core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -369,3 +385,38 @@ private fun getBlockOffset(
         String.format("getBlockOffset: Track chunk %s not in block %s", trackChunkId, blockId)
     )
 }
+
+/**
+ * Returns the stop position of a train on the signal next to a waypoint block, considering its
+ * rolling stock length.
+ *
+ * @param waypointBlock waypoint block.
+ * @param rollingStockLength length of the rolling stock.
+ * @param infra full infra.
+ * @return stop position on the signal next to the waypoint block.
+ */
+fun findStopPositionOnNextSignalConsideringRollingStock(
+    waypointBlock: PathfindingEdgeLocationId<Block>,
+    rollingStockLength: Double,
+    infra: FullInfra
+): PathfindingEdgeLocationId<Block> {
+    // To ensure that the tail of the rolling stock is not further back than the operational point
+    val minTailOffset = waypointBlock.offset.distance + rollingStockLength.meters
```

Comment:
It's more of a max limit, no? (applying the min to the 2 max values at the end)
```suggestion
    val maxTailOffset = waypointBlock.offset.distance + rollingStockLength.meters
```

Reply:
Totally right !! 💯